src/components/Popover/PopoverTrigger.tsx

```diff
@@ -1,29 +1,32 @@
 import React from 'react';
-import { ComponentWithAs, forwardRefWithAs, useForkedRef, wrapEvent } from '@reach/utils';
+import { useComposedRefs } from '@reach/utils';
 import { NativeAttributes } from '../../system';
 import { usePopoverContext } from './Popover';
 
 export type PopoverTriggerProps = NativeAttributes<'button'>;
 
-const PopoverTrigger = forwardRefWithAs<PopoverTriggerProps, 'button'>(function PopoverTrigger(
-  { as: Comp = 'button', onMouseDown, onKeyDown, ...rest },
-  forwardedRef
-) {
+const PopoverTrigger = React.forwardRef<
+  HTMLElement,
+  PopoverTriggerProps & { as: React.ElementType }
+>(function PopoverTrigger({ as: Comp = 'button', ...rest }, forwardedRef) {
   // get the ref that we should store this trigger under
   const { popoverId, triggerRef, toggle } = usePopoverContext();
 
   // merge internal + passed prop together
-  const ref = useForkedRef(triggerRef, forwardedRef);
+  const ref = useComposedRefs(triggerRef, forwardedRef);
 
   return (
     <Comp
       ref={ref}
-      onMouseDown={wrapEvent(onMouseDown, toggle)}
-      onKeyDown={wrapEvent(onKeyDown, ({ key }) => key === 'Enter' && toggle())}
+      onMouseDown={toggle}
+      onKeyDown={({ key }: { key: string }) => key === 'Enter' && toggle()}
```

Comment:
we still need to wrap them. The new method is called `composeEventHandlers` on the same package

Reply:
Good callout